### DISCUSSION TITLE
Collaboration

### DISCUSSION THREAD

**Discussion by ChristoWolf** [bchavez/Bogus#434] [General]
Hi @bchavez!

You seem to be quite busy these days, so I wanted to ask if you need any/more official contributors?
I'd love to help!

**Comment by bchavez** [comment]
Hi @ChristoWolf, thanks for the offer. Apologies for the delay. Work and life took a significant turn in the last few years. Now that both have stabilized, I've been slowly trying to get back into open-source.

Some of the most important things on my list for this project is (in rough priority):

* Reviewing and Testing:
  * https://github.com/bchavez/Bogus/pull/366
  * Requires rigorous testing; comments contain various testing scenarios and issues found

* Get the AppVeyor CI/CD system up and online again;
  * It's been a long while; and some F# parts of the build system are outdated and I don't think any PR branch will build currently
  * I'm currently exploring migrating to .NET Nuke; b/c I'm getting pretty tired of F# FAKE and the breaking changes in every release.


Of course, no obligation to help (especially after this long delay in response); but if you're still up for it, those are the areas I'd start to look at.

This isn't meant to discourage, but only maybe to set expectations on what it means on getting maintainer/write-access:
* We have about 40 million downloads and a lot of 👀 . I don't really give maintainer access right out of the gate to this repo because of the high visibility of this project. With 40M downloads, everyone from small business/hobby projects to government institutions are consuming the Bogus project. So, it's super important the source of this repo remains secure.
* Maybe, over time, if I can trust an individual, I would be open to granting write-access; provided,
  * their coding style and code quality match almost 1:1 (or better) as it exists in the current repo;
     * their PR reviews also reflect the consistent coding styles
     * share the same desire for high-quality software and are willing to go the extra mile
     * also cares a lot about public API ergonomics; sometimes it's okay to say no to a PR that wants to add a new public method.
       * https://www.amazon.com/Framework-Design-Guidelines-Conventions-Addison-Wesley-dp-0135896460/dp/0135896460/ref=dp_ob_title_bk
  * they've shown a consistently positive impact on the project and community
  * I've met them at least once, on zoom or somewhere

Hope this helps.

**Reply by ChristoWolf** [reply]
Hey @bchavez!

Happy to hear again from you, hope you are good.
I figured that there was a lot going on, and my offer still stands of course, would love to help out with the (little) time that I have!

**Reply by ChristoWolf** [reply]
Hey @bchavez!

Hope you are doing well.

Because I forgot to mention it previously: I have a masters degree in (techno-)mathematics, which I guess could be useful here.
If you are interested in any additional information about my qualifications, just let me know!

**Reply by bchavez** [reply]
Hey @ChristoWolf, that's great to hear. That's pretty cool, I haven't heard of techno-mathematics, do you have a link?

No special qualifications here to write Bogus code, just passion and motivation for high-quality, maintainable, simple, easy and clean code :)

**Reply by ChristoWolf** [reply]
I guess there is not much difference to usual mathematics, except that it includes a bit of applied math too.
Some of the more relevant things here that I learned during my studies are stochastics (e.g. probability theory) and statistics, would be happy if I could also help with those :)
